src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -130,36 +137,86 @@ case class ClusterInitValues(googleProject: GoogleProject,
                              jupyterProxySiteConf: GoogleBucketUri,
                              jupyterServerName: String,
                              proxyServerName: String,
-                             jupyterInstallExtensionScript: String,
-                             jupyterExtensionUri: GoogleBucketUri,
+                             jupyterInstallExtensionScript: GoogleBucketUri,
+                             jupyterExtensionUri: Option[GoogleBucketUri],
```

Comment:
Ah - I made this non-optional for a reason (I should have added a comment). Our init script templating code uses spray-json to substitute `$(..)` placeholders in init-actions.sh with the fields in this case class. (Aside: shapeless could also be used directly for this, it doesn't really need to be spray-json). But if `jupyterExtensionUri = None` then spray-json ignores the field, and the `$(jupyterExtensionUri)` doesn't get replaced in the init script. Then the init script crashes because it tries to interpret `$(jupyterExtensionUri)` as a command (aside #2: we probably shouldn't use bash command syntax for our template placeholders in bash scripts). Whereas if `jupyterExtensionUri = ""` then the placeholder gets substituted with the empty string, which is what we want.

TLDR: need to make this non-optional and put back the `.getOrElse("")` above, or else Bad Things will happen.

Reply:
I expanded on this a bit more in https://broadinstitute.atlassian.net/browse/GAWB-2619